etcdserver/etcdserverpb/rpc.proto

```diff
@@ -653,6 +653,9 @@ message WatchResponse {
   // The client should treat the watcher as canceled and should not try to create any
   // watcher with the same start_revision again.
   int64 compact_revision  = 5;
+  // permissionDenied is set to true if the requesting user doesn't have a
+  // read permission of watched keys.
+  bool permission_denied = 6;
```

Comment:
Thanks for your review and sorry for my late reply. I added the field (and other changes you pointed out problems) for expressing the error of permission denied. But as you say, if it can be expressed in the error code, it would be more straightfoward. I'll change the error propagation style in the next update.